Isodose/Logic/vtkSlicerIsodoseModuleLogic.cxx

```diff
@@ -683,9 +693,49 @@ void vtkSlicerIsodoseModuleLogic::CreateIsodoseSurfaces(vtkMRMLIsodoseNode* para
     vtkSlicerIsodoseModuleLogic::ISODOSE_RELATIVE_ROOT_HIERARCHY_NAME_POSTFIX :
     vtkSlicerIsodoseModuleLogic::ISODOSE_ROOT_HIERARCHY_NAME_POSTFIX;
 
-  // Setup isodose subject hierarchy folder
+  // Segmentation node data
+  vtkIdType segmentationShItemID = vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID;
+  vtkSmartPointer<vtkMRMLSegmentationNode> segmentationNode;
+  vtkSmartPointer<vtkMRMLSegmentationDisplayNode> segmentationDisplayNode;
+
+  // Setup isodose subject hierarchy folder if isodose representation is not segmentation node
   std::string isodoseFolderName = std::string(doseVolumeNode->GetName()) + isodoseName;
-  isodoseFolderItemID = shNode->CreateFolderItem(doseShItemID, isodoseFolderName);
+  if (!parameterNode->GetSegmentationFlag()) // Folder
+  {
+    isodoseFolderItemID = shNode->CreateFolderItem(doseShItemID, isodoseFolderName);
+  }
+  else // Segmentation node
+  {
+    segmentationNode = vtkSmartPointer<vtkMRMLSegmentationNode>::New();
+    std::string segmentationNodeName = scene->GenerateUniqueName(isodoseFolderName.c_str());
+    segmentationNode->SetName(segmentationNodeName.c_str());
+    scene->AddNode(segmentationNode);
+
+    // Set master representation to planar contour
+    segmentationNode->GetSegmentation()->SetMasterRepresentationName(vtkSegmentationConverter::GetSegmentationPlanarContourRepresentationName());

```

Comment:
Can you please explain in detail how you intend to use these segmentation objects, including why you use planar contour as master?

Reply:
It's an alternative option how to store isosurfaces.

the code:
` segmentationNode->SetMasterRepresentationToClosedSurface(); `
doesn't work. It doesn't show isodose surfaces on the views. It works with the planar contours.

Reply:
"Planar contours" are not suitable as master representation because they only store 2D contours and only in one specific orientation. Probably the best representation for storing isodose contours/surfaces is "closed surface".

Reply:
You can set the preferred display representations in the display node as far as I remember. That should fix the "doesn't show" part.